include/ignition/msgs/PointCloudPackedUtils.hh

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+// Inspired by
+// https://github.com/ros/common_msgs/blob/275b09a/sensor_msgs/include/sensor_msgs/point_cloud2_iterator.h
+
+#ifndef IGNITION_MSGS_POINTCLOUDPACKEDUTILS_HH_
+#define IGNITION_MSGS_POINTCLOUDPACKEDUTILS_HH_
+
+#include <ignition/msgs/pointcloud_packed.pb.h>
+
+#include <cstdarg>
+#include <string>
+#include <vector>
+#include <sstream>
+
+#include "ignition/msgs/config.hh"
+#include "ignition/msgs/detail/PointCloudPackedUtils.hh"
+
+namespace ignition
+{
+namespace msgs
+{
+/// \brief Class that can iterate over a PointCloudPacked message.
+///
+/// E.g, you create your message and reserve space for data as follows:
+/// \verbatim
+/// ignition::msgs::PointCloudPacked pcMsg;
+/// ignition::msgs::InitPointCloudPacked(pcMsg, "my_new_frame", false,
+///     {{"a", PointCloudPacked::Field::FLOAT32}});
+/// pcMsg.mutable_data()->resize(numPts * pcMsg.point_step());
+/// \endverbatim
+///
+/// For iterating over "s", you do :
```

Comment:
I'm not very familiar with this iterator class yet but should `"s"` be `"a"`?

Reply:
haha yeah it was a typo